apps/system/js/app_usage_metrics.js

```diff
@@ -695,9 +697,25 @@
     return Object.keys(this.data.apps).length === 0;
   };
 
+  // We only care about recording certain kinds of apps:
+  // - Apps pre-installed with the phone (certified)
+  // - Apps installed from the marketplace
+  UsageData.prototype.shouldTrackApp = function(app) {
+    if (!app) {
+      return false;
+    }
+
+    var manifest = app.manifest || app.updateManifest;
+    if (manifest && manifest.type === 'certified') {
```

Comment:
What if we have pre-installed apps that are not certified? There is a movement to make them privileged where we can. Could you look for gaiamobile.org in the URL as well as checking certified?  How does the homescreen decide which apps are deleteable and which are not? Is that also a certified check?